src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -880,13 +880,32 @@ private void CheckNewModifier(Symbol symbol, bool isNew, DiagnosticBag diagnosti
                                 diagnostics.Add(ErrorCode.ERR_CantChangeRefReturnOnOverride, overridingMemberLocation, overridingMember, overriddenMember);
                                 suppressAccessors = true; //we get really unhelpful errors from the accessor if the ref kind is mismatched
                             }
-                            else if (!IsValidOverrideReturnType(overridingProperty, overridingMemberType, overriddenMemberType, diagnostics))
+                            else if (overridingProperty.SetMethod is null ?
```

Comment:
Would it be easier to simply call `GetOwnOrInheritedSetMethod` here? Like we do below for doing the nullable checking?

Reply:
I will respond to this comment in https://github.com/dotnet/roslyn/pull/43797

Reply:
No, because we need to treat the presence of a set method *declared* on the property differently from the absence of one, even if one is inherited.

---
In reply to: [419707694](https://github.com/dotnet/roslyn/pull/43673#discussion_r419707694) [](ancestors = 419707694)